[CS-4539] Migrate notification api calls to RTK-Query

oidoug opened the pull request:
### Description

This PR migrates notification queries from root hub-service file to a new hub/notifications injected endpoint group, refactors the notification settings screen to use the new hooks, and adds an optimistic update for notifications preferences mutation so the switcher doesn't jump between states.

It also renames services/hub-service to hub-token-service, since it's only providing the Hub token by authenticating within the SDK.

- [x] Completes #(CS-4539)

### Checklist

- [x] Tested on a small device
- [x] Tested on iOS
- [x] Tested on Android

### Screenshots

https://user-images.githubusercontent.com/129619/204381156-fc8e510c-cda1-4422-947c-9cb980f2f65f.mp4

oidoug committed:
Moves notifications service calls already in RTK to separated files.

oidoug committed:
Refactors notifications preferences api getter and setter to rtkquery;
Adds optimistic update for notificationspreferences.

oidoug committed:
Improves and fixes mutable optimistic state update.

oidoug committed:
Fixes wrong import

oidoug committed:
Fixes return typing issue.

oidoug committed:
Fixes type import

oidoug committed:
Removes deprecated useAuthToken;
Removes deprecated functions from root hub-service;
Renames root hub-service to hub-token-service since it only provides the token now.

danibonilha commented on cardstack/src/hooks/notifications-preferences/useUpdateNotificationPreferences.ts:
```diff
@@ -14,57 +13,28 @@ export enum NotificationsOptionsStrings {
-  const { authToken, isLoading, error } = useAuthToken();
+  const {
+    data: options = [],
+    error: getError,
```
nit: getError might be misleading, if we want to use this info just as a boolean value, we can use the isError property

danibonilha commented on cardstack/src/hooks/prepaid-card/useAuthToken.ts:
```diff
@@ -1,30 +0,0 @@
-  }, [error]);
-
-  return { authToken, isLoading, error };
-};
```
So happy to finally get rid of this 🙌

danibonilha commented on cardstack/src/services/hub-token-service.ts:
```diff
@@ -0,0 +1,89 @@
+    undefined,
+    walletAddress
+  );
+};
```
Shouldn't we be storing the token on keychain ?

oidoug replied:
Hum, now that the user don't need to re-authenticate to get keychain data, it makes sense to move this as well. Probably better suited for another PR because I don't know how much work that would be. I'm adding a card.

oidoug replied:
https://linear.app/cardstack/issue/CS-4924/[debt]-move-hubs-auth-token-to-keychain

danibonilha commented on cardstack/src/services/hub/notifications/hub-notifications-api.ts:
```diff
@@ -0,0 +1,95 @@
+      query: ({ fcmToken }) => ({
+        url: `${routes.registerFCMToken}/${fcmToken}`,
+        method: 'DELETE',
+        responseHandler: response => response.text(),
```
Could this one use the `appendFMC` flag ?

oidoug replied:
For sure! Oversight from me.

danibonilha commented on cardstack/src/hooks/notifications-preferences/useUpdateNotificationPreferences.ts:
```diff
@@ -14,57 +13,28 @@ export enum NotificationsOptionsStrings {
-      if (authToken && updateItem) {
-        await setNotificationsPreferences(authToken, updateItem);
-      }
+    async (item: NotificationsPreferenceDataType, value: boolean) => {
```
nit: does value mean `isEnabled` ?, maybe we can use this as the name

danibonilha commented on cardstack/src/hooks/notifications-preferences/useUpdateNotificationPreferences.ts:
```diff
@@ -14,57 +13,28 @@ export enum NotificationsOptionsStrings {
     options,
     onUpdateOptionStatus,
-    error,
+    error: getError && 'Failed to get notificition preferences',
```
if the error is a default string maybe we could use directly on the component and just use tthe flag to conditional render it

oidoug replied:
oops, I was going to clean up this error but forgot

danibonilha reviewed (commented):
It's looking good, great cleanup

oidoug committed:
Moves error message to component to handle;
Renames some props for better clarity;
Uses appendFCMToken extraoption to unregisterFcmToken as well.

danibonilha reviewed (approved):
🚢
